ui/app/AppLayouts/Chat/stores/MessageStore.qml

```diff
@@ -206,10 +206,7 @@ QtObject {
         messageModule.addNewMessagesMarker()
     }
 
-    property bool playAnimation: {
-        if (!Global.applicationWindow.active)
```

Comment:
We still want to stop animations when the app is not in focus though. Is there a way to keep this without accessing `applicationWindow`?

Seems like you can use `Window.window.active`

Reply:
Can't be used here as the `root` is not a visual Item; however I believe this is "fixed" at the place where the call is actually used

Reply:
Yup, below in MessageView:
```qml
playAnimations: Window.window.active && root.messageStore.playAnimation
```

Reply:
(I was actually wondering about the same potential issue :)

Reply:
Ah I see. this feels a bit error prone. If someone uses the store's `playAnimation` without adding `Window.window.active`, the animations will not stop when we want.

I had an idea. Could we add a `required` property in the store called `windowActive` where components that create the store need to pass `Window.window.active`. Then, the store could use `windowActive` in `playAnimation`?

Would that work?

Reply:
I agree that leaving this property without taking window active prop into account is far from ideal because caller may be not aware that sth else should be taken into account.

Taking `windowActive` as store's param would work, but it's against the architecture we want to follow. Stores responsibility is to expose backend, UI stuff shouldn't be present there. `Window.active` is UI thing, similarly as the logic if play or not to play animation.

What seems to be wrong here right now seems to be just naming of the property. However I'm bit confused what this flag actually holds. Could it be `isChatActive`? I think it checks if chat related to this store is active, right?

With the name aligned, the responsibility of decision if animation should be played is shifted to UI:
```
playAnimations: Window.window.active && root.messageStore.isChatActive
```
The risk of incorrect usage is minimized then.

Wdyt?

Reply:
yeah that's a good compromise. I agree!